### ISSUE TITLE
Numpy install TA-Lib error

### ISSUE THREAD

**Issue by yurifontoura** [mrjbq7/ta-lib#333]
Im trying to install TA-Lib on Ubuntu 20 (of Google Cloud) and Im getting the error above. I previous installed on my local Ubuntu 20 without errors.

`  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-cnuqaw6i/TA-Lib/setup.py'"'"'; __file__='"'"'/tmp/pip-install-cnuqaw6i/TA-Lib/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-hjnhpjjc
       cwd: /tmp/pip-install-cnuqaw6i/TA-Lib/
  Complete output (5 lines):
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-cnuqaw6i/TA-Lib/setup.py", line 62, in <module>
      import numpy
  ModuleNotFoundError: No module named 'numpy'
  ----------------------------------------
  ERROR: Failed building wheel for TA-Lib
  Running setup.py clean for TA-Lib
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-cnuqaw6i/TA-Lib/setup.py'"'"'; __file__='"'"'/tmp/pip-install-cnuqaw6i/TA-Lib/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' clean --all
       cwd: /tmp/pip-install-cnuqaw6i/TA-Lib
  Complete output (5 lines):
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-cnuqaw6i/TA-Lib/setup.py", line 62, in <module>
      import numpy
  ModuleNotFoundError: No module named 'numpy'
  ----------------------------------------
  ERROR: Failed cleaning build dir for TA-Lib`

Thanks

**Comment by yurifontoura** [comment]
I already have numpy installed:

`Requirement already satisfied: numpy in ./.local/lib/python3.8/site-packages (1.19.1)
`

**Comment by mrjbq7** [comment]
Did you install numpy first?

**Comment by yurifontoura** [comment]
No. I just pip check to see if installed

**Comment by mrjbq7** [comment]
Well, the error was:

“import numpy
ModuleNotFoundError: No module named 'numpy'”

What command were you running?

pip install ta-lib

python3 setup.py install

Was it using a release or current git master?

**Comment by yurifontoura** [comment]
used: 
pip install ta-lib
git master

**Comment by yurifontoura** [comment]
I try via pip install ta-lib,

first with numpy already installed (version described above):
`/lib/python3.8/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
   17 | #warning "Using deprecated NumPy API, disable it with " \
`
and after with numpy removed:
`/ta-lib/.eggs/numpy-1.19.1-py3.8-linux-x86_64.egg/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
   17 | #warning "Using deprecated NumPy API, disable it with " \
`

**Comment by mrjbq7** [comment]
Okay, looks like on a fresh Python, if you just do ``pip install ta-lib``, it seems to not have numpy available even though it's listed as a dependency.

Solution is ``pip install numpy`` before ``pip install ta-lib``.

I'll have to look into why it's acting up that way.  

Note: this is with the last release TA-Lib-0.4.18, but I haven't tested to see if the ``setup.py`` changes on git master fix this yet.

**Comment by mrjbq7** [comment]
(The deprecation warnings are just an issue with cython using old numpy APIs, not an indication of a problem)

**Comment by vbmvbm** [comment]
i have the same issues.
ubuntu 20.04 & python3.8

 pip3 install ta-lib
Collecting ta-lib
  Downloading TA-Lib-0.4.18.tar.gz (265 kB)
Collecting numpy
  Using cached numpy-1.19.1-cp38-cp38-manylinux2010_x86_64.whl (14.5 MB)
Building wheels for collected packages: ta-lib
  Building wheel for ta-lib (setup.py) ... error

` ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0                                                                                        ] = '"'"'/tmp/pip-install-7wrk73vg/ta-lib/setup.py'"'"'; __file__='"'"'/tmp/pip-                                                                                        install-7wrk73vg/ta-lib/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(                                                                                        __file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(com                                                                                        pile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-08x08j77
       cwd: /tmp/pip-install-7wrk73vg/ta-lib/
  Complete output (80 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.8
  creating build/lib.linux-x86_64-3.8/talib
  copying talib/__init__.py -> build/lib.linux-x86_64-3.8/talib
  copying talib/test_func.py -> build/lib.linux-x86_64-3.8/talib
  copying talib/test_stream.py -> build/lib.linux-x86_64-3.8/talib
  copying talib/abstract.py -> build/lib.linux-x86_64-3.8/talib
  copying talib/test_pandas.py -> build/lib.linux-x86_64-3.8/talib
  copying talib/test_abstract.py -> build/lib.linux-x86_64-3.8/talib
  copying talib/test_data.py -> build/lib.linux-x86_64-3.8/talib
  copying talib/deprecated.py -> build/lib.linux-x86_64-3.8/talib
  copying talib/stream.py -> build/lib.linux-x86_64-3.8/talib
  running build_ext
  building 'talib._ta_lib' extension
  creating build/temp.linux-x86_64-3.8
  creating build/temp.linux-x86_64-3.8/talib
  x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -f                                                                                        wrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g                                                                                         -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-                                                                                        time -D_FORTIFY_SOURCE=2 -fPIC -I/home/ubuntu/.local/lib/python3.8/site-packages                                                                                        /numpy/core/include -I/usr/include -I/usr/local/include -I/opt/include -I/opt/lo                                                                                        cal/include -I/usr/include/python3.8 -c talib/_ta_lib.c -o build/temp.linux-x86_                                                                                        64-3.8/talib/_ta_lib.o
  In file included from /home/ubuntu/.local/lib/python3.8/site-packages/numpy/co                                                                                        re/include/numpy/ndarraytypes.h:1822,
                   from /home/ubuntu/.local/lib/python3.8/site-packages/numpy/co                                                                                        re/include/numpy/ndarrayobject.h:12,
                   from /home/ubuntu/.local/lib/python3.8/site-packages/numpy/co                                                                                        re/include/numpy/arrayobject.h:4,
                   from talib/_ta_lib.c:608:
  /home/ubuntu/.local/lib/python3.8/site-packages/numpy/core/include/numpy/npy_1                                                                                        _7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable                                                                                         it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
     17 | #warning "Using deprecated NumPy API, disable it with " \
        |  ^~~~~~~
  talib/_ta_lib.c: In function ‘__pyx_pf_5talib_7_ta_lib_348_ta_getFuncInfo’:
  talib/_ta_lib.c:37077:104: warning: passing argument 2 of ‘TA_GetFuncInfo’ fro                                                                                        m incompatible pointer type [-Wincompatible-pointer-types]
  37077 |   __pyx_v_retCode = TA_GetFuncInfo(__pyx_f_5talib_7_ta_lib___ta_getFun                                                                                        cHandle(__pyx_v_function_name), (&__pyx_v_info));
        |                                                                                                                                                                                               ~^~~~~~~~~~~~~~
        |                                                                                                                                                                                                |
        |                                                                                                                                                                                                TA_FuncInfo ** {aka struct TA_FuncInfo **}
  In file included from talib/_ta_lib.c:603:
  /usr/include/ta-lib/ta_abstract.h:211:48: note: expected ‘const TA_FuncInfo **                                                                                        ’ {aka ‘const struct TA_FuncInfo **’} but argument is of type ‘TA_FuncInfo **’ {                                                                                        aka ‘struct TA_FuncInfo **’}
    211 |                            const TA_FuncInfo **funcInfo );
        |                            ~~~~~~~~~~~~~~~~~~~~^~~~~~~~
  talib/_ta_lib.c: In function ‘__pyx_pf_5talib_7_ta_lib_350_ta_getInputParamete                                                                                        rInfo’:
  talib/_ta_lib.c:37342:127: warning: passing argument 3 of ‘TA_GetInputParamete                                                                                        rInfo’ from incompatible pointer type [-Wincompatible-pointer-types]
  37342 |   __pyx_v_retCode = TA_GetInputParameterInfo(__pyx_f_5talib_7_ta_lib__                                                                                        _ta_getFuncHandle(__pyx_v_function_name), __pyx_v_idx, (&__pyx_v_info));
        |                                                                                                                                                                                                                      ~^~~~~~~~~~~~~~
        |                                                                                                                                                                                                                       |
        |                                                                                                                                                                                                                       TA_InputParameterInfo **                                                                                         {aka struct TA_InputParameterInfo **}
  In file included from talib/_ta_lib.c:603:
  /usr/include/ta-lib/ta_abstract.h:442:68: note: expected ‘const TA_InputParame                                                                                        terInfo **’ {aka ‘const struct TA_InputParameterInfo **’} but argument is of typ                                                                                        e ‘TA_InputParameterInfo **’ {aka ‘struct TA_InputParameterInfo **’}
    442 |                                      const TA_InputParameterInfo **inf                                                                                        o );
        |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~                                                                                        ~
  talib/_ta_lib.c: In function ‘__pyx_pf_5talib_7_ta_lib_352_ta_getOptInputParam                                                                                        eterInfo’:
  talib/_ta_lib.c:37586:130: warning: passing argument 3 of ‘TA_GetOptInputParam                                                                                        eterInfo’ from incompatible pointer type [-Wincompatible-pointer-types]
  37586 |   __pyx_v_retCode = TA_GetOptInputParameterInfo(__pyx_f_5talib_7_ta_li                                                                                        b___ta_getFuncHandle(__pyx_v_function_name), __pyx_v_idx, (&__pyx_v_info));
        |                                                                                                                                                                                                                         ~^~~~~~~~~~~~~~
        |                                                                                                                                                                                                                          |
        |                                                                                                                                                                                                                          TA_OptInputParameterI                                                                                        nfo ** {aka struct TA_OptInputParameterInfo **}
  In file included from talib/_ta_lib.c:603:
  /usr/include/ta-lib/ta_abstract.h:446:74: note: expected ‘const TA_OptInputPar                                                                                        ameterInfo **’ {aka ‘const struct TA_OptInputParameterInfo **’} but argument is                                                                                         of type ‘TA_OptInputParameterInfo **’ {aka ‘struct TA_OptInputParameterInfo **’}
    446 |                                         const TA_OptInputParameterInfo                                                                                         **info );
        |                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                        ~~~^~~~
  talib/_ta_lib.c: In function ‘__pyx_pf_5talib_7_ta_lib_354_ta_getOutputParamet                                                                                        erInfo’:
  talib/_ta_lib.c:37826:128: warning: passing argument 3 of ‘TA_GetOutputParamet                                                                                        erInfo’ from incompatible pointer type [-Wincompatible-pointer-types]
  37826 |   __pyx_v_retCode = TA_GetOutputParameterInfo(__pyx_f_5talib_7_ta_lib_                                                                                        __ta_getFuncHandle(__pyx_v_function_name), __pyx_v_idx, (&__pyx_v_info));
        |                                                                                                                                                                                                                       ~^~~~~~~~~~~~~~
        |                                                                                                                                                                                                                        |
        |                                                                                                                                                                                                                        TA_OutputParameterInfo                                                                                         ** {aka struct TA_OutputParameterInfo **}
  In file included from talib/_ta_lib.c:603:
  /usr/include/ta-lib/ta_abstract.h:450:70: note: expected ‘const TA_OutputParam                                                                                        eterInfo **’ {aka ‘const struct TA_OutputParameterInfo **’} but argument is of t                                                                                        ype ‘TA_OutputParameterInfo **’ {aka ‘struct TA_OutputParameterInfo **’}
    450 |                                       const TA_OutputParameterInfo **i                                                                                        nfo );
        |                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^                                                                                        ~~~
  talib/_ta_lib.c: In function ‘__pyx_f_5talib_7_ta_lib___ta_getFuncHandle’:
  talib/_ta_lib.c:38419:126: warning: passing argument 2 of ‘TA_GetFuncHandle’ f                                                                                        rom incompatible pointer type [-Wincompatible-pointer-types]
  38419 |   __pyx_t_1 = __pyx_f_5talib_7_ta_lib__ta_check_success(__pyx_n_s_TA_G                                                                                        etFuncHandle, TA_GetFuncHandle(__pyx_v_function_name, (&__pyx_v_handle)), 0); if                                                                                         (unlikely(!__pyx_t_1)) __PYX_ERR(1, 720, __pyx_L1_error)
        |                                                                                                                                                                                                                     ~^~~~~~~~~~~~~~~~
        |                                                                                                                                                                                                                      |
        |                                                                                                                                                                                                                      TA_FuncHandle ** {aka uns                                                                                        igned int **}
  In file included from talib/_ta_lib.c:603:
  /usr/include/ta-lib/ta_abstract.h:155:52: note: expected ‘const TA_FuncHandle                                                                                         **’ {aka ‘const unsigned int **’} but argument is of type ‘TA_FuncHandle **’ {ak                                                                                        a ‘unsigned int **’}
    155 |                              const TA_FuncHandle **handle );
        |                              ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
  x86_64-linux-gnu-gcc: fatal error: Killed signal terminated program cc1
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for ta-lib
  Running setup.py clean for ta-lib
Failed to build ta-lib
DEPRECATION: Could not build wheels for ta-lib which do not use PEP 517. pip wil                                                                                        l fall back to legacy 'setup.py install' for these. pip 21.0 will remove support                                                                                         for this functionality. A possible replacement is to fix the wheel build issue                                                                                         reported above. You can find discussion regarding this at https://github.com/pyp                                                                                        a/pip/issues/8368.
Installing collected packages: ta-lib
    Running setup.py install for ta-lib ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv                                                                                        [0] = '"'"'/tmp/pip-install-7wrk73vg/ta-lib/setup.py'"'"'; __file__='"'"'/tmp/pi                                                                                        p-install-7wrk73vg/ta-lib/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open                                                                                        )(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(c                                                                                        ompile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-dkw58k                                                                                        rp/install-record.txt --single-version-externally-managed --user --prefix= --com                                                                                        pile --install-headers /home/ubuntu/.local/include/python3.8/ta-lib
         cwd: /tmp/pip-install-7wrk73vg/ta-lib/
    Complete output (80 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.8
    creating build/lib.linux-x86_64-3.8/talib
    copying talib/__init__.py -> build/lib.linux-x86_64-3.8/talib
    copying talib/test_func.py -> build/lib.linux-x86_64-3.8/talib
    copying talib/test_stream.py -> build/lib.linux-x86_64-3.8/talib
    copying talib/abstract.py -> build/lib.linux-x86_64-3.8/talib
    copying talib/test_pandas.py -> build/lib.linux-x86_64-3.8/talib
    copying talib/test_abstract.py -> build/lib.linux-x86_64-3.8/talib
    copying talib/test_data.py -> build/lib.linux-x86_64-3.8/talib
    copying talib/deprecated.py -> build/lib.linux-x86_64-3.8/talib
    copying talib/stream.py -> build/lib.linux-x86_64-3.8/talib
    running build_ext
    building 'talib._ta_lib' extension
    creating build/temp.linux-x86_64-3.8
    creating build/temp.linux-x86_64-3.8/talib
    x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g                                                                                         -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -                                                                                        g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdat                                                                                        e-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/ubuntu/.local/lib/python3.8/site-packag                                                                                        es/numpy/core/include -I/usr/include -I/usr/local/include -I/opt/include -I/opt/                                                                                        local/include -I/usr/include/python3.8 -c talib/_ta_lib.c -o build/temp.linux-x8                                                                                        6_64-3.8/talib/_ta_lib.o
    In file included from /home/ubuntu/.local/lib/python3.8/site-packages/numpy/                                                                                        core/include/numpy/ndarraytypes.h:1822,
                     from /home/ubuntu/.local/lib/python3.8/site-packages/numpy/                                                                                        core/include/numpy/ndarrayobject.h:12,
                     from /home/ubuntu/.local/lib/python3.8/site-packages/numpy/                                                                                        core/include/numpy/arrayobject.h:4,
                     from talib/_ta_lib.c:608:
    /home/ubuntu/.local/lib/python3.8/site-packages/numpy/core/include/numpy/npy                                                                                        _1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disab                                                                                        le it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
       17 | #warning "Using deprecated NumPy API, disable it with " \
          |  ^~~~~~~
    talib/_ta_lib.c: In function ‘__pyx_pf_5talib_7_ta_lib_348_ta_getFuncInfo’:
    talib/_ta_lib.c:37077:104: warning: passing argument 2 of ‘TA_GetFuncInfo’ f                                                                                        rom incompatible pointer type [-Wincompatible-pointer-types]
    37077 |   __pyx_v_retCode = TA_GetFuncInfo(__pyx_f_5talib_7_ta_lib___ta_getF                                                                                        uncHandle(__pyx_v_function_name), (&__pyx_v_info));
          |                                                                                                                                                                                               ~^~~~~~~~~~~~~~
          |                                                                                                                                                                                                |
          |                                                                                                                                                                                                TA_FuncInfo ** {aka struct TA_FuncInfo **}
    In file included from talib/_ta_lib.c:603:
    /usr/include/ta-lib/ta_abstract.h:211:48: note: expected ‘const TA_FuncInfo                                                                                         **’ {aka ‘const struct TA_FuncInfo **’} but argument is of type ‘TA_FuncInfo **’                                                                                         {aka ‘struct TA_FuncInfo **’}
      211 |                            const TA_FuncInfo **funcInfo );
          |                            ~~~~~~~~~~~~~~~~~~~~^~~~~~~~
    talib/_ta_lib.c: In function ‘__pyx_pf_5talib_7_ta_lib_350_ta_getInputParame                                                                                        terInfo’:
    talib/_ta_lib.c:37342:127: warning: passing argument 3 of ‘TA_GetInputParame                                                                                        terInfo’ from incompatible pointer type [-Wincompatible-pointer-types]
    37342 |   __pyx_v_retCode = TA_GetInputParameterInfo(__pyx_f_5talib_7_ta_lib                                                                                        ___ta_getFuncHandle(__pyx_v_function_name), __pyx_v_idx, (&__pyx_v_info));
          |                                                                                                                                                                                                                      ~^~~~~~~~~~~~~~
          |                                                                                                                                                                                                                       |
          |                                                                                                                                                                                                                       TA_InputParameterInfo                                                                                         ** {aka struct TA_InputParameterInfo **}
    In file included from talib/_ta_lib.c:603:
    /usr/include/ta-lib/ta_abstract.h:442:68: note: expected ‘const TA_InputPara                                                                                        meterInfo **’ {aka ‘const struct TA_InputParameterInfo **’} but argument is of t                                                                                        ype ‘TA_InputParameterInfo **’ {aka ‘struct TA_InputParameterInfo **’}
      442 |                                      const TA_InputParameterInfo **i                                                                                        nfo );
          |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^                                                                                        ~~~
    talib/_ta_lib.c: In function ‘__pyx_pf_5talib_7_ta_lib_352_ta_getOptInputPar                                                                                        ameterInfo’:
    talib/_ta_lib.c:37586:130: warning: passing argument 3 of ‘TA_GetOptInputPar                                                                                        ameterInfo’ from incompatible pointer type [-Wincompatible-pointer-types]
    37586 |   __pyx_v_retCode = TA_GetOptInputParameterInfo(__pyx_f_5talib_7_ta_                                                                                        lib___ta_getFuncHandle(__pyx_v_function_name), __pyx_v_idx, (&__pyx_v_info));
          |                                                                                                                                                                                                                         ~^~~~~~~~~~~~~~
          |                                                                                                                                                                                                                          |
          |                                                                                                                                                                                                                          TA_OptInputParamete                                                                                        rInfo ** {aka struct TA_OptInputParameterInfo **}
    In file included from talib/_ta_lib.c:603:
    /usr/include/ta-lib/ta_abstract.h:446:74: note: expected ‘const TA_OptInputP                                                                                        arameterInfo **’ {aka ‘const struct TA_OptInputParameterInfo **’} but argument i                                                                                        s of type ‘TA_OptInputParameterInfo **’ {aka ‘struct TA_OptInputParameterInfo **                                                                                        ’}
      446 |                                         const TA_OptInputParameterIn                                                                                        fo **info );
          |                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                        ~~~~~^~~~
    talib/_ta_lib.c: In function ‘__pyx_pf_5talib_7_ta_lib_354_ta_getOutputParam                                                                                        eterInfo’:
    talib/_ta_lib.c:37826:128: warning: passing argument 3 of ‘TA_GetOutputParam                                                                                        eterInfo’ from incompatible pointer type [-Wincompatible-pointer-types]
    37826 |   __pyx_v_retCode = TA_GetOutputParameterInfo(__pyx_f_5talib_7_ta_li                                                                                        b___ta_getFuncHandle(__pyx_v_function_name), __pyx_v_idx, (&__pyx_v_info));
          |                                                                                                                                                                                                                       ~^~~~~~~~~~~~~~
          |                                                                                                                                                                                                                        |
          |                                                                                                                                                                                                                        TA_OutputParameterInf                                                                                        o ** {aka struct TA_OutputParameterInfo **}
    In file included from talib/_ta_lib.c:603:
    /usr/include/ta-lib/ta_abstract.h:450:70: note: expected ‘const TA_OutputPar                                                                                        ameterInfo **’ {aka ‘const struct TA_OutputParameterInfo **’} but argument is of                                                                                         type ‘TA_OutputParameterInfo **’ {aka ‘struct TA_OutputParameterInfo **’}
      450 |                                       const TA_OutputParameterInfo *                                                                                        *info );
          |                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                        ~^~~~
    talib/_ta_lib.c: In function ‘__pyx_f_5talib_7_ta_lib___ta_getFuncHandle’:
    talib/_ta_lib.c:38419:126: warning: passing argument 2 of ‘TA_GetFuncHandle’                                                                                         from incompatible pointer type [-Wincompatible-pointer-types]
    38419 |   __pyx_t_1 = __pyx_f_5talib_7_ta_lib__ta_check_success(__pyx_n_s_TA                                                                                        _GetFuncHandle, TA_GetFuncHandle(__pyx_v_function_name, (&__pyx_v_handle)), 0);                                                                                         if (unlikely(!__pyx_t_1)) __PYX_ERR(1, 720, __pyx_L1_error)
          |                                                                                                                                                                                                                     ~^~~~~~~~~~~~~~~~
          |                                                                                                                                                                                                                      |
          |                                                                                                                                                                                                                      TA_FuncHandle ** {aka u                                                                                        nsigned int **}
    In file included from talib/_ta_lib.c:603:
    /usr/include/ta-lib/ta_abstract.h:155:52: note: expected ‘const TA_FuncHandl                                                                                        e **’ {aka ‘const unsigned int **’} but argument is of type ‘TA_FuncHandle **’ {                                                                                        aka ‘unsigned int **’}
      155 |                              const TA_FuncHandle **handle );
          |                              ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
    x86_64-linux-gnu-gcc: fatal error: Killed signal terminated program cc1
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sy                                                                                        s, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-7wrk73vg/ta-lib/set                                                                                        up.py'"'"'; __file__='"'"'/tmp/pip-install-7wrk73vg/ta-lib/setup.py'"'"';f=getat                                                                                        tr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"                                                                                        ', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' instal                                                                                        l --record /tmp/pip-record-dkw58krp/install-record.txt --single-version-external                                                                                        ly-managed --user --prefix= --compile --install-headers /home/ubuntu/.local/incl                                                                                        ude/python3.8/ta-lib Check the logs for full command output.
`

**Comment by yurifontoura** [comment]
On AWS and Azure installation goes fine. But for Google (GCP) I've been had some install issues. I still do not know how to solve it. Bytheway one solution was to use docker instalation and now it seems work fine. As I found a way to avoid error I will close it.

**Comment by mrjbq7** [comment]
Closing all installation issues, if someone has a specific problem please open a new one.

**Comment by MagicEthan** [comment]
Here's how I solved this:
Ubuntu 18.04(default python 3.6.9), and I need to install python3.7.0 and TA-Lib==0.4.17. Firstly I installed Python3.7.0 without set the prefix, and the executable file was in /usr/local/bin as default, then I got the same problem even if I've installed numpy. The tricky thing is that if you install talib without setting the version, the latest version works just fine, and I also found that TA-Lib>=0.4.19 goes well, but TA-Lib<=0.4.18 can't work. (Numpy==1.18.5, Pandas=0.24.2)
I reinstalled python3.7.0 with . /configure --prefix=/usr/local/python3 (remember to mkdir), and all those things including pip were there. Then running pip3 install TA-Lib=0.4.17, it works.
